### ISSUE TITLE
Feature Request: Support for Aurora Parallel Queries

### ISSUE THREAD

**Issue by chrisdewar** [terraform-providers/terraform-provider-aws#5977]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

See feature here:
https://aws.amazon.com/blogs/aws/new-parallel-query-for-amazon-aurora/

It would seem that most of the groundwork is already in place to support this feature, however some validation of the `engine_mode` argument seems to be preventing this from going through. Maybe that is an oversimiplification of the problem though...

### New or Affected Resource(s)

* aws_rds_cluster

### Potential Terraform Configuration

```hcl
resource "aws_rds_cluster" "default" {
  cluster_identifier      = "aurora-cluster-demo"
  availability_zones      = ["us-west-2a", "us-west-2b", "us-west-2c"]
  database_name           = "mydb"
  master_username         = "foo"
  master_password         = "bar"
  backup_retention_period = 5
  preferred_backup_window = "07:00-09:00"
  engine                  = "aurora"
  engine_mode             = "parallelquery"
}
```

Current error:
```
Error: aws_rds_cluster.persistence-platform-master: expected engine_mode to be one of [provisioned serverless], got parallelquery
```

### References

* https://www.terraform.io/docs/providers/aws/r/rds_cluster.html#engine_mode
* https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_rds_cluster.go#L128
* https://aws.amazon.com/blogs/aws/new-parallel-query-for-amazon-aurora/
* https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBCluster.html

**Comment by bflad** [comment]
Hi @chrisdewar 👋 Thanks for submitting this! Indeed it looks like we just needed to open up the `engine_mode` validation for this new value. I have submitted a pull request with updated validation, testing, and documentation here: #5980

**Comment by bflad** [comment]
This has been released in [version 1.38.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1380-september-26-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
